src/compiler/checker.ts

```diff
@@ -18547,6 +18550,12 @@ namespace ts {
                 type;
         }
 
+        function removeUndefinedType(type: Type): Type {

```

Comment:
`getTypeWithFacts(type, TypeFacts.NEUndefined)` should suffice for this.

Reply:
cool.